Sources/HBLambdaTest/maths.swift

```diff
@@ -14,20 +14,57 @@
 
 import AWSLambdaEvents
 import AWSLambdaRuntime
+import Hummingbird
 import HummingbirdFoundation
 import HummingbirdLambda
 import NIO
+import Logging
 
 struct DebugMiddleware: HBMiddleware {
-    func apply(to request: HBRequest, next: HBResponder) -> EventLoopFuture<HBResponse> {
-        request.logger.debug("\(request.method) \(request.uri)")
-        request.logger.debug("\(request.apiGatewayV2Request)")
-        return next.respond(to: request)
+    typealias Context = MathsHandler.Context
+
+    func apply(
+        to request: HBRequest,
+        context: Context,
+        next: any HBResponder<Context>
+    ) async throws -> HBResponse {
+        context.logger.debug("\(request.method) \(request.uri)")
+        if let apiGatewayRequest = context.apiGatewayRequest {
+            context.logger.debug("\(apiGatewayRequest)")
+        } else {
+            context.logger.debug("No APIGatewayV2Request")
+        }
+
+        return try await next.respond(to: request, context: context)
     }
 }
 
 @main
 struct MathsHandler: HBLambda {
+    struct Context: HBLambdaRequestContext {
+        typealias Event = APIGatewayRequest
+
+        let apiGatewayRequest: APIGatewayRequest?
+        var coreContext: HBCoreRequestContext
+
+        init(applicationContext: HBApplicationContext, source: some RequestContextSource, logger: Logger) {
+            self.apiGatewayRequest = nil
+            self.coreContext = HBCoreRequestContext(
+                applicationContext: applicationContext,
+                eventLoop: source.eventLoop,
+                logger: logger
+            )
+        }
+
+        init(_ event: APIGatewayRequest, applicationContext: HBApplicationContext, source: some RequestContextSource, logger: Logger) {
```

Comment:
It's a bit frustrating you have to write both `init` functions. The only way I can think of getting around this would be have a base protocol `_HBBaseRequestContext` that requires access to the  `coreContext` but with no `init` function requirement. Then setup `HBRequestContext` to conform to `_HBBaseRequestContext` and add the `init` from `Channel` requirement. And then have `HBLambdaRequestContext` that conforms to `_HBBaseRequestContext` and adds a `init` from lambda event requirement.

Reply:
What do you think should we implement that before merging this? We could get rid of the RequestContextSource as we are have full control over the parameters being sent to the init function

Reply:
That makes sense to me, glad I presented the code as-is to you because that's a great solution. Let's make a HB PR for that first

Reply:
Ok I'll set something up in HB